app/src/main/scala/org/alephium/explorer/service/TokenService.scala

```diff
@@ -0,0 +1,279 @@
+      dc: DatabaseConfig[PostgresProfile]
+  ): Future[ArraySeq[TokenInfo]] =
+    run(listTokenInfosQuery(tokens))
+
+  def listFungibleTokenMetadata(tokens: ArraySeq[TokenId])(implicit
+      dc: DatabaseConfig[PostgresProfile]
+  ): Future[ArraySeq[FungibleTokenMetadata]] =
+    run(listFungibleTokenMetadataQuery(tokens))
+
+  def listNFTMetadata(tokens: ArraySeq[TokenId])(implicit
+      dc: DatabaseConfig[PostgresProfile]
+  ): Future[ArraySeq[NFTMetadata]] =
+    run(listNFTMetadataQuery(tokens))
+
+  def listNFTCollectionMetadata(addresses: ArraySeq[Address.Contract])(implicit
+      dc: DatabaseConfig[PostgresProfile]
+  ): Future[ArraySeq[NFTCollectionMetadata]] =
+    run(listNFTCollectionMetadataQuery(addresses))
+
+  def listAddressTokens(address: Address, pagination: Pagination)(implicit
+      dc: DatabaseConfig[PostgresProfile]
+  ): Future[ArraySeq[TokenId]] =
+    run(listAddressTokensAction(address, pagination))
+
+  def listAddressTokenTransactions(address: Address, token: TokenId, pagination: Pagination)(
+      implicit
+      ec: ExecutionContext,
+      dc: DatabaseConfig[PostgresProfile]
+  ): Future[ArraySeq[Transaction]] =
+    run(getTokenTransactionsByAddress(address, token, pagination))
+
+  def listAddressTokensWithBalance(address: Address, pagination: Pagination)(implicit
+      ec: ExecutionContext,
+      dc: DatabaseConfig[PostgresProfile]
+  ): Future[ArraySeq[(TokenId, U256, U256)]] =
+    run(listAddressTokensWithBalanceAction(address, pagination))
+
+  def listTokenWithoutInterfaceId()(implicit
+      ec: ExecutionContext,
+      dc: DatabaseConfig[PostgresProfile]
+  ): Future[ArraySeq[TokenId]] = {
+    run(listTokenWithoutInterfaceIdQuery())
+  }
+
+  def listContractWithoutInterfaceId()(implicit
+      ec: ExecutionContext,
+      dc: DatabaseConfig[PostgresProfile]
+  ): Future[ArraySeq[Address.Contract]] = {
+    run(listContractWithoutInterfaceIdQuery())
+  }
+
+  // Interface Id is store after the metadata is stored in case it fail in the middle of the
+  // metadata insert. As we rely on the empyt interface id to know which token to update.
+  def fetchAndStoreTokenMetadata(token: TokenId, blockflowClient: BlockFlowClient)(implicit
+      ec: ExecutionContext,
+      dc: DatabaseConfig[PostgresProfile]
+  ): Future[Unit] = {
+    logger.debug(s"Update token $token")
+    blockflowClient.guessTokenStdInterfaceId(token).flatMap { interfaceIdOpt =>
+      (interfaceIdOpt match {
+        case Some(StdInterfaceId.FungibleToken) =>
+          blockflowClient.fetchFungibleTokenMetadata(token).flatMap {
+            case Some(metadata) =>
+              run(insertFungibleTokenMetadata(token, metadata)).map(_ => ())
+            case None => Future.successful(())
+          }
+        case Some(StdInterfaceId.NFT) =>
+          blockflowClient.fetchNFTMetadata(token).flatMap {
+            case Some(metadata) =>
+              run(insertNFTMetadata(token, metadata)).map(_ => ())
```

Comment:
Question: Is it ok that this query and `updateTokenInterfaceId` below are non-transactional? Same question for `fetchAndStoreTokenMetadata`.

Reply:
I guess I simply didn't thought of using `transactionnaly`. If the second fail it's not that a problem, we will simply re-fetch the token and it will be ignored by the ` DO NOTHING`. But better to use a `transactionnaly`. I'll Fix that.

Reply:
done here [c703c1f](https://github.com/alephium/explorer-backend/pull/486/commits/c703c1fc90f989dfedab8ca6f6c6c6e67f4fc61c)